packages/shared/lib/clients/big-query.client.ts

```diff
@@ -0,0 +1,72 @@
+import { BigQuery } from '@google-cloud/bigquery';
+import logger from '../logger/console.js';
+import { isCloud, isLocal, getEnv } from '../utils/utils.js';
+
+type Row = Record<string, string | number | undefined>;
+
+class BigQueryClient {
+    private client: BigQuery;
+    private datasetName: string;
+    private tableName: string;
+
+    constructor() {
+        this.client = new BigQuery();
+        this.datasetName = 'dataset';
+        this.tableName = 'script_runs';
+    }
+
+    private async createDataSet() {
+        const dataset = this.client.dataset(this.datasetName);
+        const [exists] = await dataset.exists();
+        if (!exists) {
+            await this.client.createDataset(this.datasetName);
+        }
+    }
+
+    private async createTable() {
+        const table = this.client.dataset(this.datasetName).table(`${getEnv()}_${this.tableName}`);
+        const [exists] = await table.exists();
+        if (!exists) {
+            await table.create({
+                schema: {
+                    fields: [
+                        { name: 'executionType', type: 'STRING' },
+                        { name: 'internalConnectionId', type: 'INTEGER' },
+                        { name: 'connectionId', type: 'STRING' },
+                        { name: 'accountId', type: 'INTEGER' },
+                        { name: 'scriptName', type: 'STRING' },
+                        { name: 'scriptType', type: 'STRING' },
+                        { name: 'environmentId', type: 'INTEGER' },
+                        { name: 'providerConfigKey', type: 'STRING' },
+                        { name: 'status', type: 'STRING' },
+                        { name: 'syncId', type: 'STRING' },
+                        { name: 'content', type: 'STRING' },
+                        { name: 'runTime', type: 'INTEGER' }, // in seconds
+                        { name: 'timestamp', type: 'INTEGER' }
```

Comment:
what is timestamp? date of creation of the row, or start of the sync or end of the sync?